task 도메인생성

goodcodings opened the pull request:
Signed-off-by: Lee Byoung Seong <dev.bslee@gmail.com>

goodcodings committed:
task 도메인생성

Signed-off-by: Lee Byoung Seong <dev.bslee@gmail.com>

ahastudio commented on app/src/main/java/com/codesoom/assignment/domain/Task.java:
```diff
@@ -0,0 +1,72 @@
+    private LocalDateTime createdAt;
+    private LocalDateTime updatedAt;
+
+    private Task(TaskBuilder taskBuilder) {
```
이 방식은 이펙티브 자바에서 소개하고 있는데, 여기서는 몇 가지를 고려해 보세요.

1. 일반적인 (GoF의) 빌더 패턴이라면 의존성을 반대 방향으로 만듭니다. 인터페이스 중심(즉, 구현은 여럿)으로 다루기 위한 패턴이기 때문이죠.
2. 만약 빌더 패턴을 의도한 게 아니라면 ParameterObject로 정리하는 게 낫습니다. https://refactoring.com/catalog/introduceParameterObject.html

객체를 고치는 경우라면 빌더로 들어가지 않는 게 일반적입니다. 지금은 몇 가지 불안정한 가정에 의존하고 있습니다:

1. 해당 ID의 객체가 이미 있음을 전제로 하고 있습니다(만약 없다면 강제 생성으로 처리해야 할까요?).
2. 기존 객체를 완전히 대체하는 걸 전제로 하고 있습니다(예를 들어 생성일시를 유지하거나, 상태 변환 가능 여부를 확인해야 한다면?).
3. if문으로 분기해야 할 정도로 두 경우가 크게 다르다는 걸 전제하고 있습니다(실제로 코드를 보면 그렇지 않아 다수의 중복이 있죠).

goodcodings committed:
[Week1] 빌더패턴 제거

goodcodings committed:
[Week1] 빌더패턴 제거

goodcodings committed:
[Week1] Todo REST api 만들기
1. DIP를 최대한 지키기 위해 역할과 나눴습니다
   - 의존 주입을하는 AppConfig 클래스를 생성했습니다.
2. 에릭 에반스의 ddd sample을 참조하여 패키지 구조를 잡았습니다.
3. task의 빌더 패턴을 다시 만들었고 title 변경시 task 객체의 자율성을 위해 updateTitle 메소드를  생성했습니다

Lee-Byoung-Seong committed:
[Week1] Todo REST api 만들기
1. DIP를 최대한 지키기 위해 역할과 구현을 나눴습니다
- 의존 주입을하는 AppConfig 클래스를 생성했습니다.
2. 에릭 에반스의 ddd sample을 참조하여 패키지 구조를 잡았습니다.
3. task의 빌더 패턴을 다시 만들었고 title 변경시 task 객체의 자율성을 위해 updateTitle 메소드를 생성했습니다
4. httpHandler를 구현하여 클라이언트로 부터 요청 받고 응답하는 코드를 생성했습니다
5. Todo 생성까지 구현했습니다
6. 각 메소드에 맞는 처리를 하기 위해 post,get,put,delete클래스를 생성했습니다.

Lee-Byoung-Seong committed:
[Week1] Todo REST api 만들기
1. DIP를 최대한 지키기 위해 역할과 구현을 나눴습니다
- 의존 주입을하는 AppConfig 클래스를 생성했습니다.
2. 에릭 에반스의 ddd sample을 참조하여 패키지 구조를 잡았습니다.
3. task의 빌더 패턴을 다시 만들었고 title 변경시 task 객체의 자율성을 위해 updateTitle 메소드를 생성했습니다
4. httpHandler를 구현하여 클라이언트로 부터 요청 받고 응답하는 코드를 생성했습니다
5. Todo 생성까지 구현했습니다
6. 각 메소드에 맞는 처리를 하기 위해 post,get,put,delete클래스를 생성했습니다.

Lee-Byoung-Seong committed:
[Week1] Todo REST api 만들기
1. DIP를 최대한 지키기 위해 역할과 구현을 나눴습니다
- 의존 주입을하는 AppConfig 클래스를 생성했습니다.
2. 에릭 에반스의 ddd sample을 참조하여 패키지 구조를 잡았습니다.
3. task의 빌더 패턴을 다시 만들었고 title 변경시 task 객체의 자율성을 위해 updateTitle 메소드를 생성했습니다
4. httpHandler를 구현하여 클라이언트로 부터 요청 받고 응답하는 코드를 생성했습니다
5. Todo 생성까지 구현했습니다
6. 각 메소드에 맞는 처리를 하기 위해 post,get,put,delete클래스를 생성했습니다.
